Adding pagination to disputes grid

eburgos opened the pull request:
Refers to #62

Disputes grid with pagination. I hope it helps.

Regards,

eburgos committed:
Adding pagination to disputes grid. Refers to #62

# Conflicts:
#	src/Components/DisputesTable/GridContent/index.js

eburgos committed:
Fix ESLint and add some tests

Changed code to meet ESLint issues. Added a test to not
decrease overall coverage.

n1c01a5 commented on src/Components/DisputesTable/GridContent/index.js:
```diff
@@ -8,13 +8,35 @@ import { truncateText } from '../../../helpers/truncateText'
+      currentPage: 1
+    }
+    this.paginateHandler = this.paginateHandler.bind(this)
+    this.paginateInputHandler = this.paginateInputHandler.bind(this)
```
I think you don't need `bind()` see https://medium.com/shoutem/react-to-bind-or-not-to-bind-7bf58327e22a

eburgos replied:
Inside those method I'm settings this.state. If I don't bind then I won't have this available (would be undefined). Please let me know if you consider a workaround.

n1c01a5 commented on src/Components/DisputesTable/GridContent/index.js:
```diff
@@ -8,13 +8,35 @@ import { truncateText } from '../../../helpers/truncateText'
     this.props.getDataDisputes()
   }
 
+  paginateHandler (event) {
```
could use the functional syntax for your own function like `paginateHandler = e => (this.setState({ currentPage: Number(e.target.dataset.pageNumber) }))`

n1c01a5 commented on src/Components/DisputesTable/GridContent/index.js:
```diff
@@ -8,13 +8,35 @@ import { truncateText } from '../../../helpers/truncateText'
+    this.setState({
+      currentPage: Number(event.target.value)
+    })
+  }
```
idems

n1c01a5 commented on src/Components/DisputesTable/GridContent/index.js:
```diff
@@ -32,6 +54,14 @@ class GridContent extends Component {
+    })
+    const currentRows = list.slice(indexOfFirstRow, indexOfLastRow)
+
+    const totalPages = Math.ceil(list.length / rowsPerPage) || 1
```
can you create a function like `const currentRows = (list, indexOfLastRow, indexOfFirstRow)` in the `utils` folder

eburgos replied:
I couldn't find a utils folder. Perhaps you meant "helpers" ?

n1c01a5 commented:
and can you hide the "paginator" if there are not enough disputes

eburgos committed:
Add requested changes on #93

Pagination now shows only if there is more than 1 page.
Added story to storybook showing dispute table with no pagination.
Styled the paginator a bit to make it look more like the theme.
Added test on the grid not showing pagination.

eburgos commented:
@n1c01a5 Fixed most of the stuff you requested. 
1) I couldn't find that "utils" folder, perhaps you meant the "helpers" one
2) Regarding the .bind(this) I need to use "this" inside the function so I can't skip that bind. Unless there is another way of achieving this that I'm not aware.

n1c01a5 commented:
@eburgos yes it's the `helpers` folder and yes I think we can skip it but it's about the refactoring. We can see that later. Tnx for your contribution.
